pkgs/applications/science/biology/star/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, zlib }:
+
+stdenv.mkDerivation rec {
+  name = "star";
+  version = "2.5.3a";
+
+  src = fetchFromGitHub {
+    repo = "STAR";
+    owner = "alexdobin";
+    rev = version;
+    sha256 = "1fd9xl7i1zxgsxn2qf6gz8s42g2djm29qmp6qb35d8nnxh8ns54x";
+  };
+
+  buildInputs = [ zlib ];
+
+  buildPhase = ''
+    cd source
```

Comment:
Why not use `sourceRoot`? See https://nixos.org/nixpkgs/manual/#ssec-unpack-phase.

Reply:
Sure, what's the idiomatic way to do this? `sourceRoot = "source/source";`?

Reply:
Just `sourceRoot = "source"` should be fine.

Reply:
I had tried `sourceRoot = "source"` already, but it didn't work. I might be missing something entirely obvious as I'm migrating packages from Guix, and there are some subtle differences.

Note that STAR has a `source` directory, which shows up inside Nix-created `source` after unpacking. This is the derivation I'm using, which works fine:

```nix
{ stdenv, fetchFromGitHub, zlib }:

stdenv.mkDerivation rec {
  name = "star-${version}";
  version = "2.5.3a";

  src = fetchFromGitHub {
    repo = "STAR";
    owner = "alexdobin";
    rev = version;
    sha256 = "1fd9xl7i1zxgsxn2qf6gz8s42g2djm29qmp6qb35d8nnxh8ns54x";
  };

  sourceRoot = "source/source";
  
  postPatch = "sed 's:/bin/rm:rm:g' -i Makefile";
  
  buildInputs = [ zlib ];
  
  buildPhase = "make STAR STARlong";

  installPhase = ''
    mkdir -p $out/bin
    cp STAR STARlong $out/bin
  '';
  
  meta = with stdenv.lib; {
    description = "Spliced Transcripts Alignment to a Reference";
    homepage = https://github.com/alexdobin/STAR;
    license = licenses.gpl3Plus;
    platforms = platforms.linux;
    maintainers = [ maintainers.arc ];
  };
}
```

Reply:
Looks alright :+1: